lib/services/platform-service.ts

```diff
@@ -482,8 +482,34 @@ export class PlatformService implements IPlatformService {
 	}
 
 	public deployOnEmulator(platform: string, buildConfig?: IBuildConfig): IFuture<void> {
-		this.$options.emulator = true;
-		return this.deployOnDevice(platform, buildConfig);
+		platform = platform.toLowerCase();
+
+		if (this.$options.availableDevices) {
+			return (() => {
+				let exec = require("child_process").exec;
```

Comment:
Will it be faster if we require this once (e.g. check if undefined and then require)?